### DISCUSSION TITLE
Pros and cons vs Isolated (WordPress) Block Editor by Automattic

### DISCUSSION THREAD

**Discussion by earthlingdavey** [puckeditor/puck#241] [General]
Hi measured & contributors, first up, Puck looks awesome. There's a high chance I'll end up using it as a user facing editor in a react app. So TY!

But, during my research, I've found a project that I think is very similar: [Isolated (WordPress) Block Editor by Automattic](https://github.com/Automattic/isolated-block-editor)

If I don't know which route to explore, if possible would you mind adding to the comparisson?

| Feature | Puck | Isolated Block Editor |
| --- | --- | --- |
| Self host | ✅ | ✅ |
| FOSS | ✅ | ✅ |
| Works on small screens | ❌ | ✅ |
| Drop into Next.JS | ✅ | ❌[*](https://github.com/Automattic/isolated-block-editor/issues/55) |
| Built for JS devs  | ✅ | ❌ |
| Nested components | ❌ | ✅ |
| Comes with component library | ✅ | ✅ |

Did you folks consider building upon existing work, I would appreciate your insight. Please keep up the good work and understand this question is meant sincerely.

Cheers - Davey

**Comment by chrisvxd** [comment]
Hi @EarthlingDavey! Great question and thanks for putting together the table.

Firstly, some comments on your table - 

* Works on small screens - we're planning to resolve this soon (but it's not currently tracked here).
* Nested components - we do support this via the [DropZone API](https://github.com/measuredco/puck#dropzones)

Updated table:

| Feature | Puck | Isolated Block Editor |
| --- | --- | --- |
| Self host | ✅ | ✅ |
| FOSS | ✅ | ✅ |
| Works on small screens | ⏱️ | ✅ |
| Drop into Next.JS | ✅ | ❌[*](https://github.com/Automattic/isolated-block-editor/issues/55) |
| Built for JS devs  | ✅ | ❌ |
| Nested components | ✅ | ✅ |
| Comes with component library | ❌ | ✅ |
| License | MIT | GPL |

Finally, in terms of building upon existing work, we couldn't find suitable options at the time. We needed something lightweight we could integrate into React-based commercial projects for our clients. For example, isolated block editor uses GPL, which makes it impossible for proprietary use (which we need). Puck uses the extremely permissive MIT license, allowing commercial use.

**Comment by earthlingdavey** [comment]
Really nice to see you have small viewports planned :) The target audience for my project is like > 60% mobile. 

And good to hear that puck has nesting already. I hadn't even thought about GPL, it's a shame because it probably means they're missing out on features getting pushed back upstream from agencies like yours.

Thanks for the updated table. I see I was confused about Pucks internal components being a core library, it's no dealbreaker for me.

**Comment by chrisvxd** [comment]
Great! If you want a more direct line of engagement, feel free to join [our Discord server](https://discord.gg/D9e4E3MQVZ)

**Comment by chrisvxd** [comment]
I'm going to move this to discussions.
